backend-test/Controllers/CourseControllerTests.hs

```diff
@@ -6,17 +6,88 @@ Module that contains the tests for the functions in the Course Controller module
 -}
 
 module Controllers.CourseControllerTests
-( courseControllerTestSuite ) where
+( courseControllerTestSuite
+) where
 
 import Config (runDb)
-import Controllers.Course (index)
+import Control.Applicative (Alternative(..))
+import Control.Monad (unless)
+import Controllers.Course (index, retrieveCourse)
 import qualified Data.ByteString.Lazy.Char8 as BL
+import qualified Data.Map as Map
+import Data.Maybe (fromMaybe)
 import qualified Data.Text as T
 import Database.Persist.Sqlite (SqlPersistM, insert_)
 import Database.Tables (Courses(..))
 import Happstack.Server (rsBody)
 import Test.HUnit (Test(..), assertEqual)
-import TestHelpers (clearDatabase, runServerPart)
+import TestHelpers (clearDatabase, runServerPart, runServerPartWithQuery)
+
+-- | List of test cases as (input course name, course data, expected JSON output)
+retrieveCourseTestCases :: [(String, T.Text, Map.Map String String, String)]
```

Comment:
Make the type of this map `Map.Map T.Text T.Text`. If we use the [`OverloadedStrings` compiler extension](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/overloaded_strings.html) you can keep the string literals as written, and the compiler will automatically interpret these as `T.Text` values rather than `String` values. I believe the `courseography.cabal` file has enabled `OverloadedStrings` globally, but may not have done so for the tests, in which case you should add it to the configuration in that file.